doc/articles/migrating-to-uno-5.md

```diff
@@ -0,0 +1,226 @@
+
+Use the `Control` class instead.
+
+#### `ViewHelper.MainScreenScale` and `ViewHelper.IsRetinaDisplay` are removed in iOS and macOS
+
+Use the `DisplayInformation` class instead.
+
+#### WebView.NavigateWithHttpRequestMessage parameter type is now `Windows.Web.Http.HttpRequestMessage` instead of `System.Net.Http.HttpRequestMessage`
+
+Use the new types instead.
+
+#### `OnVerticalOffsetChanged` and `OnHorizontalOffsetChanged` are now private instead of virtual protected
+
+Use the related events instead.
+
+#### `FrameBufferHost` and `GtkHost` constructors that takes `args` are now removed. `args` were already unused
+
+You can remove the last argument of the constructor invocation. The parameters are read by the host directly.
+
+#### `RegisterDefaultStyleForType` methods were removed
+
+This legacy method was deprecated in Uno 3.x
+
+#### Xaml code generator was generating an always-null-returning FindResource method. This method is no longer generated
+
+This legacy method was deprecated in Uno 3.x
+
+#### The type `Windows.Storage.Streams.InMemoryBuffer` is removed
+
+Use `Windows.Storage.Streams.Buffer` instead.
+
+#### `ContentPropertyAttribute.Name` is now a field to match UWP
+
+This change has no effect on Controls behavior.
+
+#### Remove `FontWeightExtensions` and `CssProviderExtensions`
+
+`Uno.UI.Runtime.Skia.GTK.UI.Text.FontWeightExtensions` and `Uno.UI.Runtime.Skia.GTK.Extensions.Helpers.CssProviderExtensions` don't exist in UWP/WinUI. So they are made internal.
+
+#### Change `GtkHost`, `WpfHost`, `FrameBufferHost` namespaces
+
+`GtkHost`, `WpfHost`, and `FrameBufferHost` are now `Uno.UI.Runtime.Skia.Gtk.GtkHost`, `Uno.UI.Runtime.Skia.Wpf.WpfHost`, and `Uno.UI.Runtime.Skia.Linux.FrameBuffer.FrameBufferHost` instead of `Uno.UI.Runtime.Skia.GtkHost`, `Uno.UI.Skia.Platform.WpfHost`, and `Uno.UI.Runtime.Skia.FrameBufferHost`, respectively.
+
+#### Change `RenderSurfaceType` namespace
+
+There used two be two `RenderSurfaceType`s, `Uno.UI.Runtime.Skia.RenderSurfaceType` (for Gtk) and `Uno.UI.Skia.RenderSurfaceType` (for Skia). They are now `Uno.UI.Runtime.Skia.Gtk.RenderSurfaceType` and `Uno.UI.Runtime.Skia.Wpf.RenderSurfaceType` respectively.
+
+#### `Panel`s no longer measure or arrange any children in `MeasureOverride` or `ArrangeOverride`, respectively
+
+`Panel`s used to measure and arrange the first child in `MeasureOverride` or `ArrangeOverride`, respectively. This is no longer the case. Now, to match WinUI, `Panel`s just return an empty size in `MeasureOverride`, and the `finalSize` as is in `ArrangeOverride`. You should override these layout-override methods in `Panel`-derived subclasses instead.
+
+#### Remove `Windows.UI.Xaml.Controls.NavigationView` in Uno.WinUI
+
+Uno.WinUI used to have NavigationView both in `Microsoft.UI.Xaml.Controls` and `Windows.UI.Xaml.Controls` namespaces. It's now removed from `Windows.UI.Xaml.Controls` and kept only in `Microsoft.UI.Xaml.Controls` namespace.
```

Comment:
```suggestion
#### `FrameBufferHost` and `GtkHost` constructors that take `args` are now removed. `args` were already unused

You can remove the last argument of the constructor invocation. The parameters are read by the host directly.

#### `RegisterDefaultStyleForType` methods were removed

This legacy method was deprecated in Uno 3.x

#### Xaml code generator was generating an always-null-returning FindResource method. This method is no longer generated

This legacy method was deprecated in Uno 3.x

#### The type `Windows.Storage.Streams.InMemoryBuffer` is removed

Use `Windows.Storage.Streams.Buffer` instead.

#### `ContentPropertyAttribute.Name` is now a field to match UWP

This change has no effect on Control's behavior.

#### Remove `FontWeightExtensions` and `CssProviderExtensions`

`Uno.UI.Runtime.Skia.GTK.UI.Text.FontWeightExtensions` and `Uno.UI.Runtime.Skia.GTK.Extensions.Helpers.CssProviderExtensions` don't exist in UWP/WinUI. So they are made internal.

#### Change `GtkHost`, `WpfHost`, `FrameBufferHost` namespaces

`GtkHost`, `WpfHost`, and `FrameBufferHost` are now `Uno.UI.Runtime.Skia.Gtk.GtkHost`, `Uno.UI.Runtime.Skia.Wpf.WpfHost`, and `Uno.UI.Runtime.Skia.Linux.FrameBuffer.FrameBufferHost` instead of `Uno.UI.Runtime.Skia.GtkHost`, `Uno.UI.Skia.Platform.WpfHost`, and `Uno.UI.Runtime.Skia.FrameBufferHost`, respectively.

#### Change `RenderSurfaceType` namespace

There used two be two `RenderSurfaceType`s, `Uno.UI.Runtime.Skia.RenderSurfaceType` (for Gtk) and `Uno.UI.Skia.RenderSurfaceType` (for Skia). They are now `Uno.UI.Runtime.Skia.Gtk.RenderSurfaceType` and `Uno.UI.Runtime.Skia.Wpf.RenderSurfaceType` respectively.

#### `Panel`s no longer measure or arrange any children in `MeasureOverride` or `ArrangeOverride`, respectively

`Panel`s used to measure and arrange the first child in `MeasureOverride` or `ArrangeOverride`, respectively. This is no longer the case. Now, to match WinUI, `Panel`s just return an empty size in `MeasureOverride`, and the `finalSize` as is in `ArrangeOverride`. You should override these layout-override methods in `Panel`-derived subclasses instead.

#### Remove `Windows.UI.Xaml.Controls.NavigationView` in Uno.WinUI

Uno.WinUI used to have NavigationView both in `Microsoft.UI.Xaml.Controls` and `Windows.UI.Xaml.Controls` namespaces. It's now removed from `Windows.UI.Xaml.Controls` and kept only in `Microsoft.UI.Xaml.Controls` namespace.
```